api/v1alpha1/datadogagent_default_test.go

```diff
@@ -620,7 +621,7 @@ func TestDefaultDatadogAgentSpecAgent(t *testing.T) {
 				UseExtendedDaemonset: NewBoolPointer(false),
 				Image: &ImageConfig{
 					Name:        "gcr.io/datadog/agent:6.26.0",
-					Tag:         defaultAgentImageTag,
+					Tag:         defaulting.AgentLatestVersion,
```

Comment:
The name in the line above doesn't correspond to "latest".

Reply:
I keep `Latest` in the name, because it correspond to the "latest released" version. 
Do you think we should find another name?

Reply:
What I meant in this case is that the `Name` field in the line above already includes the version (`6.26.0`) and it does not correspond to `defaulting.AgentLatestVersion`. I think that `:6.26.0` shouldn't be there.

Reply:
Yes, I now understand this is a weird behaviour of the defaulting, that doesn't check if the `Name` contains the tag. I will try to fix it

Reply:
I have updated the defaulting logic to only set the "default tag" if the tag is not already present in the image name.
you can review the change in this [commit](https://github.com/DataDog/datadog-operator/pull/355/commits/46d0b8b03bad88baac20797e279d4be6f7533b29)